.storybook/lucid-docs-addon/register.js

```diff
@@ -0,0 +1,245 @@
+import '@storybook/addon-options/register';
+import React from 'react';
+import addons from '@storybook/addons';
+import _ from 'lodash';
+import ExampleCode from './ExampleCode';
+import PropTypes from './PropTypes';
+import packageJson from '../../package.json';
+
+class CodePanel extends React.Component {
+	constructor(...args) {
+		super(...args);
+		this.state = { code: '' };
+		this.onSource = this.onSource.bind(this);
+	}
+
+	onSource(code) {
+		if (!_.isString(code)) {
+			return this.setState({
+				code: '',
+			});
+		}
+
+		this.setState({
+			code,
+		});
+	}
+
+	componentDidMount() {
+		const { channel, api } = this.props;
+		// Listen to the notes and render it.
+		channel.on('lucid-docs-source', this.onSource);
+
+		// Clear the current notes on every story change.
+		this.stopListeningOnStory = api.onStory(() => {
+			this.onSource('');
+		});
+	}
+
+	render() {
+		const { code } = this.state;
+
+		if (_.isEmpty(code)) {
+			return null;
+		}
+
+		return (
+			<div style={{ width: '100%' }}>
+				<ExampleCode code={code} hasCodepen packageJson={packageJson} />
+			</div>
+		);
+	}
+
+	// This is some cleanup tasks when the CodePanel is unmounting.
+	componentWillUnmount() {
+		if (this.stopListeningOnStory) {
+			this.stopListeningOnStory();
+		}
+
+		this.unmounted = true;
+		const { channel, api } = this.props;
+		channel.removeListener('lucid-docs-source', this.onSource);
+	}
+}
+
+class PropsPanel extends React.Component {
+	constructor(...args) {
+		super(...args);
+		this.state = { props: null };
+		this.onDisplayProps = this.onDisplayProps.bind(this);
+	}
+
+	onDisplayProps(propsJSON) {
+		if (!_.isString(propsJSON)) {
+			return this.setState({
+				props: null,
+			});
+		}
+
+		try {
+			this.setState({
+				props: JSON.parse(propsJSON),
+			});
+		} catch (err) {
+			this.setState({
+				props: null,
+			});
+		}
+	}
+
+	componentDidMount() {
+		const { channel, api } = this.props;
+		// Listen to the notes and render it.
+		channel.on('lucid-docs-display-props', this.onDisplayProps);
+
+		// Clear the current notes on every story change.
+		this.stopListeningOnStory = api.onStory(() => {
+			this.onDisplayProps('');
+		});
+	}
+
+	componentWillUnmount() {
+		if (this.stopListeningOnStory) {
+			this.stopListeningOnStory();
+		}
+
+		this.unmounted = true;
+		const { channel, api } = this.props;
+		channel.removeListener('lucid-docs-display-props', this.onDisplayProps);
+	}
+
+	render() {
+		return (
+			<div
+				style={{
+					backgroundColor: 'white',
+					width: '100%',
+					overflow: 'auto',
+					padding: 10,
+					position: 'relative',
+				}}
+			>
+				<PropTypes props={this.state.props} />
+			</div>
+		);
+	}
+}
+
+class SettingsPanel extends React.Component {
+	constructor(...args) {
+		super(...args);
+
+		const [props] = args;
+		const urlState = props.api.getUrlState();
```

Comment:
might be a little more defensive with this: `const urlState = _.get(props, 'api.getUrlState', _.noop)();`